meson.build

```diff
@@ -1,13 +1,15 @@
 project('mpv',
         'c',
         license: ['GPL2+', 'LGPL2.1+'],
-        version: files('./VERSION'),
+        version: files('./MPV_VERSION'),
         meson_version: '>=0.62.0',
         default_options: [
             'backend_max_links=16',
             'buildtype=debugoptimized',
             'b_lundef=false',
-            'c_std=c11',
+            'c_std=gnu11',
+            'cpp_std=c++20',
+            'cpp_eh=default',
```

Comment:
> I think that's the way to go if only C++ requires it. By the way, which part of mpv requires __STRICT_ANSI__ to be undefined?

I missed this comment. Only this line requires https://github.com/mpv-player/mpv/blob/5158a3d779ed5b6bcd41b4a0573847abc08dcb66/video/out/filter_kernels.c#L326

See comment above, `j1` is not declared otherwise on mingw build.

Reply:
> Since j1 is POSIX

Well, it is supported by Windows CRT too https://learn.microsoft.com/cpp/c-runtime-library/reference/bessel-functions-j0-j1-jn-y0-y1-yn and without any preconditions, it is just that mingw headers are hiding it while they shouldn't.

Reply:
Sorry, I checked patched headers.

MinGW-w64 explicitly hides them behind `#ifndef __STRICT_ANSI__` See: https://github.com/mingw-w64/mingw-w64/blob/55ced9c1c9aca9ebab1e6236c507dbb09c33ca31/mingw-w64-headers/crt/math.h#L261-L311

MSYS2 has a patch to loosen this. See: https://github.com/msys2/MINGW-packages/blob/45f37941c39acf3cdf08c243d3dc5fce5385ca08/mingw-w64-headers-git/0001-Allow-to-use-bessel-and-complex-functions-without-un.patch

Since it is not upstream version, the only way to access either `j1` or `_j1` in mingw is by undefining `__STRICT_ANSI__`.

> Windows exposes _j1, which is not POSIX because it has an underscore.

Well, obviously. But one can `#define j1 _j1` or `-D_CRT_DECLARE_NONSTDC_NAMES` to use it. I don't need POSIX here.

Reply:
> I guess we're stuck with undefining then.

Will be fixed upstream. https://sourceforge.net/p/mingw-w64/mailman/message/58782943/

Reply:
Fixed by https://github.com/mingw-w64/mingw-w64/commit/74875157b0387c71eac09327d021f599a9be193a